src/movies.js

```diff
@@ -1,16 +1,66 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+function getAllDirectors(moviesArray) {
+  const directors = moviesArray.map(function (name) {
+    return name.director;
+  });
+  return directors;
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+  let count = 0;
+  moviesArray.filter(function (drama) {
+    if (
+      drama.director === "Steven Spielberg" &&
+      drama.genre.includes("Drama")
+    ) {
+      count++;
+    }
+  });
+  return count;
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+function scoresAverage(moviesArray) {
+  if (moviesArray.length === 0) {
+    return 0;
+  }
+
+  let avgrate = moviesArray.reduce(function (accumulator, currentValue) {
+    if (!currentValue.score) {
+      currentValue = 0;
+      return accumulator + currentValue;
+    }
+    return accumulator + currentValue.score;
+  }, 0);
+
+  let result = avgrate / moviesArray.length;
+  return Number(result.toFixed(2));
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore(moviesArray) {}
+function dramaMoviesScore(moviesArray) {
+  let dramaMovies = moviesArray.filter((element) => {
+    return element.genre.includes("Drama") && element.score;
+  });
+
+  if (dramaMovies.length === 0) {
+    return 0;
+  } else {
+    let dramaScore =
+      dramaMovies.reduce((acc, cValue) => {
+        if (typeof cValue.score !== "number") {
+          cValue.score = 0;
+          return acc + cValue.score;
+        } else {
+          return acc + cValue.score;
+        }
+      }, 0) / dramaMovies.length;
+    return Number(dramaScore.toFixed(2));
```

Comment:
In that one you could have reused the previous function to calculate the average